src/main/java/org/sefglobal/scholarx/controller/ProgramController.java

```diff
@@ -53,4 +56,38 @@ public Mentor applyAsMentor(@PathVariable long id,
         long profileId = 1; // TODO: Get the profileId from headers
         return programService.applyAsMentor(id, profileId, mentor);
     }
+
+    @GetMapping("/{id}/mentor")
+    @ResponseStatus(HttpStatus.OK)
+    public Mentor getLoggedInMentor(@PathVariable long id) throws ResourceNotFoundException {
+        long profileId = 1;  // TODO: Get the profileId from headers
+        return programService.getLoggedInMentor(id, profileId);
+    }
+
+    @GetMapping("/{id}/mentees")
+    @ResponseStatus(HttpStatus.OK)
+    public List<Mentee> getMenteesOfAMentor(@PathVariable long id)
+            throws ResourceNotFoundException {
+        long profileId = 1;  // TODO: Get the profileId from headers
+        return programService.getAllMenteesOfAMentor(profileId, id);
+    }
+
+    @PutMapping("/{id}/application")
+    @ResponseStatus(HttpStatus.OK)
+    public Mentor updateApplicationAndPrerequisites(@PathVariable long id,
+                                                    @Valid @RequestBody Mentor mentor)
+            throws ResourceNotFoundException {
+        long profileId = 1;  // TODO: Get the profileId from headers
+        return programService.updateApplicationAndPrerequisites(profileId, id, mentor);
+    }
+
+    @PutMapping("/{id}/mentees/{menteeId}/state")
+    @ResponseStatus(HttpStatus.OK)
+    public Mentee approveOrRejectMentee(@PathVariable long id,
+                                        @PathVariable long menteeId,
+                                        @Valid @RequestBody EnrolmentState state)
+            throws ResourceNotFoundException, BadRequestException {
```

Comment:
Instead of `EnrolmentState` use a boolean: `isApproved`

Reply:
Is there a possibility of getting a null request body from the user?

Reply:
no and it's better to use a boolean